### ISSUE TITLE
Refreshing Token On XML HTTP Requests

### ISSUE THREAD

**Issue by mikelunn** [dzimchuk/azure-ad-b2c-asp-net-core#7]
@dzimchuk First of all thank you for this repository. I've referred to this repository to fill the gaps that I feel are completely left out of the Azure B2C documentation. One aspect is refresh tokens. The re-authentication filter that you have implemented works great. But I notice that it only handles full page refreshes. It doesn't seem to work for xml http requests. As an example, the MVC client renders a page to the browser with a script on the page that makes an async call back to the MVC client. If the token has expired, the proxy class will throw the reauthentication required exception which the filter would handle. However, like I said this only seems to work on full page post backs and doesn't work for xml http requests. Any idea on how to handle the token refresh for xml http requests that need to be proxied to the web api?

**Comment by mikelunn** [comment]
Just noticed your comments in startup.cs about offline_access. Perhaps that's why i'm unable to obtain refresh tokens.

**Comment by dzimchuk** [comment]
@mikelunn Yes, offline_access is a standard scope that indicates you want Azure AD to include a refresh token in the response.

The sample app is a regular server side web application so yes the re-authentication filter is only applicable there. On the client side a similar logic can be implemented (although I haven't tried). See https://github.com/AzureAD/microsoft-authentication-library-for-js.

**Comment by mikelunn** [comment]
@dzimchuk I've used msal.js as well. I'm basically trying to have my MVC app serve mini-spa applications depending on the route. I've set up a middleware in .net core to add the authorization header with the bearer token before proxying the "api" call to the WebAPI. I guess it's worth trying to incorporate msal.js into the client side code to see if I can obtain the refresh token in the browser. I'll give msal.js a shot. I'll let you know if it works.
